stan/math/prim/mat/fun/gp_exp_quad_cov.hpp

```diff
@@ -72,8 +76,8 @@ inline
  * @tparam T_sigma type of sigma
  * @tparam T_l type of std::vector of length scale
  *
- * @param x std::vector of elements that can be used in square distance.
- *    This function assumes each element of x is the same size.
+ * @param x std::vector of Eigen column vectors of scalars.
+ *    This function assumes each column vector of x is the same size.
  * @param sigma standard deviation
  * @param length_scale std::vector length scale
  * @return squared distance
```

Comment:
[doc fix]
Can you update the `@throw` doc? This should be enough (please check my wording / write it in a way that seems right):
```
@throw std::illegal_argument if x has one or more elements and the size of length scale does not match the number of rows of the elements of x
```